gpkit/__init__.py

```diff
@@ -95,6 +96,50 @@ def __call__(self, arg):
 
 enable_units()
 
+VECTORIZATION = ()
+MODELS = []
+MODELNUMS = []
+MODELNUM_LOOKUP = defaultdict(int)
+
+
+def begin_variable_naming(model):
+    global MODELS, MODELNUMS
+    MODELS.append(model)
+    num = MODELNUM_LOOKUP[model]
+    MODELNUMS.append(num)
+    MODELNUM_LOOKUP[model] += 1
+    return num, (tuple(MODELS), tuple(MODELNUMS))
+
+
+def end_variable_naming():
+    global MODELS, MODELNUMS
+    MODELS = MODELS[:-1]
+    MODELNUMS = MODELNUMS[:-1]
```

Comment:
They're not assigned to the last elements here; on the contrary, `[:-1]` just chops off the last element.